src/index.ts

```diff
@@ -336,17 +336,24 @@ const normalizeOptions = async (
   setSilent(options.silent)
 
   const input = options.entryPoints
-  if (input) {
-    options.entryPoints = await glob(input)
-  } else {
+
+  if (!input) {
     throw new PrettyError(`No input files, try "tsup <your-file>" instead`)
   }
 
-  // Ensure entry exists
-  if (!options.entryPoints || options.entryPoints.length === 0) {
-    throw new PrettyError(`Cannot find ${input}`)
+  if (Array.isArray(input)) {
+    options.entryPoints = await glob(input)
+    // Ensure entry exists
+    if (!options.entryPoints || options.entryPoints.length === 0) {
+      throw new PrettyError(`Cannot find ${input}`)
+    } else {
+      log('CLI', 'info', `Building entry: ${options.entryPoints.join(', ')}`)
+    }
   } else {
-    log('CLI', 'info', `Building entry: ${options.entryPoints.join(', ')}`)
+    if (Object.keys(input).length === 0) {
+      throw new PrettyError(`Cannot find ${JSON.stringify(input)}`)
+    }
+    log('CLI', 'info', `Building entry: ${JSON.stringify(input)}`)
```

Comment:
you should properly check if the files exist when `input` is an object, now you're just checking if it's empty